kong/runloop/balancer.lua

```diff
@@ -483,18 +483,19 @@ do
 
 
   ------------------------------------------------------------------------------
-  -- Finds and returns an upstream entity. This function covers
-  -- caching, invalidation, db access, et al.
-  -- @param upstream_name string.
-  -- @return upstream table, or `false` if not found, or nil+error
+  -- Implements a simple dictionary with all upstream-ids indexed
+  -- by their name.
+  -- @return The upstreams dictionary (a map with upstream names as string keys
+  -- and upstream entity tables as values), or nil+error
   get_all_upstreams = function()
-    local upstreams_dict, err = singletons.cache:get("balancer:upstreams", nil,
+    local opts = { neg_ttl = 10 }
```

Comment:
We could avoid extra GC cycles if we didn't allocate this table inside this relatively hot function.

Reply:
ah yes, for a second I misread where this function was being called from and thought it wasn't worth it — you're right, fixed